### DISCUSSION TITLE
Pass different domains to useFetch() for server and client calls

### DISCUSSION THREAD

**Discussion by dhwid** [nuxt/nuxt#20109] [Questions]
Hi Nuxt Community!

My goal is to pass different domains to useFetch() for the server and the client calls, without [Nuxt Server API Endpoints](https://nuxt.com/docs/guide/directory-structure/server#api-routes). For example the client URL starts with `public domain over https`,  while the server URL starts with `localhost over http`. 

I was not able to reproduce exact scenario on stackblitz but the result is the same because URLs for the server and for the client are different  https://stackblitz.com/edit/github-drhg3c?file=app.vue

```javascript
console.log('before fetch');
const resp = await useFetch(process.server ? urlServer : urlClient); //different URLs but the same response data
console.log('after fetch', resp.data?.value);
```

On the server side the behaviour is as expected. But on client side, the behaviour is suprising:

_Network:_
![image](https://user-images.githubusercontent.com/26313423/230363506-5af704ca-57ab-4a65-99c3-98afeba44b6c.png)

_Console:_
![image](https://user-images.githubusercontent.com/26313423/230363088-ad287c82-43d7-495e-a248-b49b51265c1c.png)

The client (browser) fetches data, it takes 51 ms but `useFetch()` returns `null` after just 2 ms (trying to use cached data from server, but URL is different?) and the real client's response seems not to be used at all. Finally nuxt returns hydration mismatch warning because the data on the client side is null .  

Is `useFetch()` designed to support described case? 

P.S If SSR is disabled averything looks fine:

_Console:_
![image](https://user-images.githubusercontent.com/26313423/230365887-ab5ed23c-4fa0-4c01-a71a-f244cdd7e868.png)

**Comment by ClementGre** [comment]
Hi, I had the same issue and de-duplicated the requests by specifying the ``key`` option of the ``useFetch`` function:

From https://nuxt.com/docs/api/composables/use-fetch:
> ``key``: a unique key to ensure that data fetching can be properly de-duplicated across requests, if not provided, it will be automatically generated based on URL and fetch options

You can then write something like this:
```js
let {data, error} = await useFetch(backend_host + path, {
    key: path,
    ...
})
```
Maybe you should consider adding more information like request parameters in the key to prevent too much de-duplication issues, see #21532 for an example and a related issue it could lead to.
